stocks/stocks.py

```diff
@@ -1,7 +1,155 @@
\ No newline at end of file
+    return "This is an example stock app"
+
+def get_stock_quote(symbol):
+    """
+    Get real-time quote data for a stock given symbol
+    """
+    r = requests.get(STOCK_QUOTE_URL.format(token=FINNHUB_TOKEN, symbol=symbol))
+    response = r.json()
+    return response
+    """
+        {
+            "c": 261.74,
+            "h": 263.31,
+            "l": 260.68,
+            "o": 261.07,
+            "pc": 259.45,
+            "t": 1582641000 
+        }
+    """
+
+def validate_stock_symbol(symbol):
+    """
+    Make sure symbol is trackable
+    """
+    r = requests.get(STOCK_PROFILE_URL.format(token=FINNHUB_TOKEN, symbol=symbol))
+    response = r.json()
+    if response: 
+        return response.get("name")
+    return False
+
+
+
+def calculate_percent_change(response, average_cost):
+    """
+    Calculate percent change in stock compared to average cost
+    """
+    current_price = response["c"]
+    difference = current_price - average_cost
+    percent_difference = round(difference / average_cost * 100, 2)
+    return percent_difference
+    
+
+def insert_stock_tracker(stock_id, average_cost, percent, increase, decrease):
+    """
+    Insert tracked stock average cost, percent, increase, decrease into database
+    """
+    con = sql.connect(config.DATABASE) 
+    cur = con.cursor()
+    
+    cur.execute("SELECT id FROM stock_tracker WHERE stock_id=?", [stock_id]) 
+    if cur.fetchone():
+        cur.execute("DELETE FROM stock_tracker WHERE stock_id=?", [stock_id])
+        con.commit()
+    cur.execute("INSERT INTO stock_tracker (average_cost, percent, increase, decrease, stock_id) VALUES(?,?,?,?,?)", (average_cost, percent, increase, decrease, stock_id))
+    con.commit()
+
+
+def filter_tracked_stocks_details(symbol, detailed):
+    con = sql.connect(config.DATABASE) 
+    cur = con.cursor()
+        
+    if symbol:
+        cur.execute("SELECT * FROM stock_tracker WHERE stock_id = (SELECT id from stock WHERE symbol=?)", [symbol]) 
+        tracked_stocks = [cur.fetchone()]
+    else:
+        cur.execute("SELECT * FROM stock_tracker")
+        tracked_stocks = cur.fetchall()
+
+    if any(stock is None for stock in tracked_stocks):
+        return 
+
+    tracked_stocks_list = []
+    for stock_details in tracked_stocks:
+        cur.execute("SELECT symbol, name FROM stock WHERE id=?", [stock_details[6]]) 
+        stock_profile = cur.fetchone()
+        symbol = stock_profile[0]
+        name = stock_profile[1]
+        average_cost = stock_details[1]
+        percent = stock_details[2]
+        increase = True if stock_details[3] else False
+        decrease = True if stock_details[4] else False
+        last_modified = stock_details[5]
+
+        if detailed:
+            response = get_stock_quote(symbol)
+            if not response:
+                return jsonify({"Error": "You are tracking an invalid stock symbol: {symbol}".format(symbol=symbol)}), 404        
```

Comment:
Does this work? I don't think it's getting returned properly upstream

Reply:
Alternative with f-string:
```
"error": f"You are tracking an invalid stock symbol: {symbol}"
```

Lowercase error to standardize response

Reply:
works but ill switch to f-string, looks better